pkg/queue/bounded_queue.go

```diff
@@ -60,11 +69,18 @@ func (q *BoundedQueue) StartConsumers(num int, consumer func(item interface{}))
 			defer q.stopWG.Done()
 			for {
 				select {
-				case item := <-q.items:
-					atomic.AddInt32(&q.size, -1)
-					consumer(item)
 				case <-q.stopCh:
 					return
+				default:
+					for _, items := range q.items {
+						select {
+						case item := <-items:
```

Comment:
sooooo, this iterates over the priority queue and starting at the highest priority channel, checks if there's an item. If there is, it consumes it. This repeats for all channels.

it feels wrong

Reply:
It is wrong - hot cycle, no blocking.

Reply:
i'm gonna try using a priority queue but I feel like the all the extra syncing required will kill performance vs hot cycling.

I'll benchmark both and give you the numbers

Reply:
> all the extra syncing required

this is a common misconception - do you think channels are free from syncing? There are many cases where a simple mutex is a lot more efficient than achieving the same functionality with channels.